matchbox_socket/src/webrtc_socket/native.rs

```diff
@@ -188,110 +187,113 @@ impl Messenger for NativeMessenger {
-            ),
         }
+        .compat() // Required for compatibility between wasm & native
+        .await
     }
 
     async fn peer_loop(peer_uuid: PeerId, handshake_meta: Self::HandshakeMeta) -> PeerId {
-        let (mut to_peer_message_rx, data_channels, mut trickle_fut, mut peer_disconnected) =
-            handshake_meta;
-
-        assert_eq!(
-            data_channels.len(),
-            to_peer_message_rx.len(),
-            "amount of data channels and receivers differ"
-        );
-
-        let mut message_loop_futs: FuturesUnordered<_> = data_channels
-            .iter()
-            .zip(to_peer_message_rx.iter_mut())
-            .map(|(data_channel, rx)| async move {
-                while let Some(message) = rx.next().await {
-                    trace!("sending packet {message:?}");
-                    let message = message.clone();
-                    let message = Bytes::from(message);
-                    if let Err(e) = data_channel.send(&message).await {
-                        error!("error sending to data channel: {e:?}")
+        async {
+            let (mut to_peer_message_rx, data_channels, mut trickle_fut, mut peer_disconnected) =
+                handshake_meta;
+
+            assert_eq!(
+                data_channels.len(),
+                to_peer_message_rx.len(),
+                "amount of data channels and receivers differ"
+            );
+
+            let mut message_loop_futs: FuturesUnordered<_> = data_channels
+                .iter()
+                .zip(to_peer_message_rx.iter_mut())
+                .map(|(data_channel, rx)| async move {
+                    while let Some(message) = rx.next().await {
+                        trace!("sending packet {message:?}");
+                        let message = message.clone();
+                        let message = Bytes::from(message);
+                        if let Err(e) = data_channel.send(&message).await {
+                            error!("error sending to data channel: {e:?}")
+                        }
                     }
-                }
-            })
-            .collect();
+                })
+                .collect();
 
-        loop {
-            select! {
-                _ = peer_disconnected.next() => break,
+            loop {
+                select! {
+                    _ = peer_disconnected.next() => break,
 
-                _ = message_loop_futs.next() => break,
-                // TODO: this means that the signalling is down, should return an
-                // error
-                _ = trickle_fut => continue,
+                    _ = message_loop_futs.next() => break,
+                    // TODO: this means that the signalling is down, should return an
+                    // error
+                    _ = trickle_fut => continue,
+                }
             }
-        }
 
-        peer_uuid
+            peer_uuid
+        }
+        .compat() // Required for compatibility between wasm & native
```

Comment:
```suggestion
        .compat() // Required for running on non-tokio async executors without panicking
```
This is not just about cross platform compatibility

Reply:
Changed to "Required to run tokio async executors using other runtimes"